lebedigital/raw_data_processing/metadata_extraction/mixture_metadata_extraction.py

```diff
@@ -1,142 +1,190 @@
-                    "Wasserzementwert" : exceltodf.iloc[7,2],
-                    "Konsistenz" : exceltodf.iloc[7,7],
-                    "Sieblinie n. DIN 1045" : exceltodf.iloc[8,2],
-                    "Körnungsziffer" : exceltodf.iloc[8,8],
-                    "Vol Leim/Zuschlag" : exceltodf.iloc[10,10]
-                    }
-            word = "Zusatzstoff"
-        specimeninfo = pd.Series(specimeninfo)
-        specimeninfo = specimeninfo.fillna("---") # fill nans
-        exceltodf.rename(columns=exceltodf.iloc[17], inplace = True) # set column header
-        exceltodf["Stoffart"] = exceltodf["Stoffart"].str.strip()  # remove whitespace
-        exceltodf["Sonstiges/Bemerkungen"] = exceltodf["Sonstiges/Bemerkungen"].str.strip()  # remove whitespace
-        
-        # Some files have multiple entries of Zusatzstoffe, which are not further labeled, so this adds 
-        # the type of Zusatzstoff
-        howmanyzusatzstoffe = [True if i == word else False for i in exceltodf["Stoffart"] ]
-        indexzusatzstoff = [i for i in range(len(howmanyzusatzstoffe)) if howmanyzusatzstoffe[i] == True]
-        for i in indexzusatzstoff:
-            exceltodf.iloc[i,0] += " " + str(exceltodf.iloc[i,1])
+        # save data from excelsheet into pandas dataframe
+        exceltodf = excelfile[sheet]
      
-        # create new dataframe with only relevant data and chose Index column
-        relevant_data = exceltodf.iloc[20:,[0,2,4,8]]
-        relevant_data = relevant_data.set_index("Stoffart")
-        
-        # replace NaN 
-        relevant_data[relevant_data.columns[-1]] = relevant_data[relevant_data.columns[-1]].fillna("---")
-        relevant_data = relevant_data.drop(nan)
+        # create empty dictionary for metadata
+        metadata = {}
+
+        # the layout of the excel table can vary, the indices of labels are not 
+        # always the same; that's why: find now the indices of the labels
+        labelidx = {}
+        labelcolumn = exceltodf.iloc[:,0]  # select first column (containing labels)
+        print(labelcolumn[0])
+        for i in range(len(labelcolumn)):
+            labelcolumn[i] = str(labelcolumn[i]).strip()  # remove whitespace
+            labelidx[labelcolumn[i]] = i  # fill dictionary
+        #print(labelidx)
+
+        # check for missing and additional labels
+
+
+
+        ############### E X T R A C T I O N #############
+
+        # get date and time (always the same position)
+        metadata['operator_date'] = str(exceltodf.columns[9])[:10]
+
+        # operator name (always the same position)
+        metadata['tester_name'] = exceltodf.iloc[0,10]
+
+        # name of specimen
+        idx = labelidx["Bezeichnung der Proben:"]
+        metadata['specimen_name'] = exceltodf.iloc[idx,3] #4
+
+        # water cement ratio ("Wasserzementwert")
+        idx = labelidx["Wasserzementwert"]
+        metadata["watercementratio"] = exceltodf.iloc[idx,2] #7
```

Comment:
what are these numbers #7 as comments?

Reply:
I would recompute it.

Reply:
Just use the ratio between mass of water and mass of cement both given in the excel sheet separately. That should evaluate to a numeric value somewhere between 0.2 and 0.7 and should agree with the entry in this cell (without the additional sometimes weird comments that we currently do not use anyway).